internal/service/advancedcluster/fw_data_source_advanced_clusters.go

```diff
@@ -0,0 +1,183 @@
+package advancedcluster
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/id"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+const (
+	advancedClustersDatasourceName = "advanced_clusters"
+)
+
+var _ datasource.DataSource = &advancedClustersDS{}
+var _ datasource.DataSourceWithConfigure = &advancedClustersDS{}
+
+type advancedClustersDS struct {
+	config.DSCommon
+}
+
+func PluralDataSource() datasource.DataSource {
+	return &advancedClustersDS{
+		DSCommon: config.DSCommon{
+			DataSourceName: advancedClustersDatasourceName,
+		},
+	}
+}
+
+func (d *advancedClustersDS) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{ // required by hashicorps terraform plugin testing framework: https://github.com/hashicorp/terraform-plugin-testing/issues/84#issuecomment-1480006432
+				DeprecationMessage: "Please use each cluster's id attribute instead",
+				Computed:           true,
+			},
+			"project_id": schema.StringAttribute{
+				Required: true,
+			},
+			"results": schema.ListNestedAttribute{
+				Computed: true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: advancedClusterDSAttributes(),
```

Comment:
Not certain if there is a risk here: `advancedClusterDSAttributes()` defines `project_id` as required, which would not be correct in the case of computed results for this data source. As an example, in stream instance a boolean is passed to handle this difference but be able to reuse the schema: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/streaminstance/data_source_stream_instance.go#L35